src/ripple/app/main/GRPCServer.cpp

```diff
@@ -0,0 +1,270 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/app/main/GRPCServer.h>
+
+namespace ripple {
+
+template <class Request, class Response>
+GRPCServerImpl::CallData<Request, Response>::CallData(
+              rpc::v1::XRPLedgerAPIService::AsyncService& service,
+              grpc::ServerCompletionQueue& cq,
+              Application& app,
+              BindListener<Request, Response> bind_listener,
+              Handler<Request, Response> handler,
+              RPC::Condition required_condition,
+              Resource::Charge load_type)
+          : 
+              service_(service),
+              cq_(cq),
+              status_(PROCESSING),
+              app_(app),
+              iter_(boost::none),
+              aborted_(false),
+              responder_(&ctx_),
+              bind_listener_(bind_listener),
+              handler_(handler),
+              required_condition_(required_condition),
+              load_type_(load_type)
+{
+      //Bind a listener. When a request is received, "this" will be returned
+      //from CompletionQueue::Next
+      bind_listener_(
+              service_, &ctx_, &request_, &responder_, &cq_, &cq_, this);
+}
+
+template<class Request, class Response>
+std::shared_ptr<Processor> GRPCServerImpl::CallData<Request, Response>::clone()
+{
+  return std::static_pointer_cast<Processor>(
+          std::make_shared<CallData<Request, Response>>(
+              service_,
+              cq_,
+              app_,
+              bind_listener_,
+              handler_,
+              required_condition_,
+              load_type_));
```

Comment:
Cloning something should probably use the copy constructor. If something has a zero argument `clone` function is should have a copy constructor as well. I'd probably also return a raw pointer to the derived type rather than a shared pointer from `clone` (C++ has covariant return types, so it's OK for `clone` to return the derived type if we use the raw pointer).

Reply:
Keeping the clone method as is. See the below thread about using shared pointers